src/oc_erchef/apps/oc_chef_wm/src/chef_wm_actions_queue_monitoring.erl

```diff
@@ -0,0 +1,472 @@
+    {noreply, State}.
+
+
+%%-------------------------------------------------------------
+%% INFO
+handle_info(reset_dropped_since_last_check, State) ->
+    {noreply,State#state{dropped_since_last_check = 0}};
+handle_info({'EXIT', From, Reason}, #state{worker_process = WorkerPid,
+                                           sync_response_process = SyncPid} = State) ->
+    % Check to see if the EXIT came from our worker
+    case From == WorkerPid of
+        true ->
+              % check if we need to reply to a sync_check_current_state() call
+              case SyncPid of
+                  undefined -> ok;
+                  Pid -> gen_server:reply(Pid, ok)
+              end,
+              % clear the worker_process, allowing for future workers to start
+              {noreply, State#state{worker_process = undefined,
+                                    sync_response_process = undefined}};
+        false ->
+            lager:warning("Unknown process exit detected in Queue Monitor ~p ~p", [From, Reason]),
+          {noreply, State}
+    end;
+handle_info({MaxLength, N, AtCap}, State) ->
+    % a successful check of the max queue length and current queue length
+    % just update the state and carry on
+    {noreply,State#state{
+               max_length = MaxLength,
+               last_recorded_length = N,
+               queue_at_capacity = AtCap,
+               dropped_since_last_check = 0
+              }};
+% guard against starting more than one worker process via the
+% match to worker_process = undefined.
+handle_info(status_ping, #state{worker_process = undefined,
+                                dropped_since_last_check = Dropped} = State) ->
+    ParentPid = self(),
+    Pid = spawn_link(fun () -> check_current_queue_state(ParentPid, Dropped) end),
+    {noreply, State#state{worker_process=Pid}};
+handle_info(status_ping, State) ->
+    lager:info("Queue monitor check still running, skipping next check"),
+    {noreply, State};
+handle_info(Info, State) ->
+    lager:debug("Unknown info: ~p", [Info]),
+    {noreply, State}.
+
+terminate(_Reason, #state{timer=undefined}) ->
+    ok;
+terminate(_Reason, #state{timer=Timer}) ->
+    {ok, cancel} = timer:cancel(Timer),
+    ok.
+
+
+code_change(_OldVsn, State, _Extra) ->
+    {ok, State}.
+
+
+%%-------------------------------------------------------------
+%% private
+
+% start the timer that updates stats
+-spec start_update_timer() -> timer:tref().
+start_update_timer() ->
+    Interval = envy:get(oc_chef_wm, rabbitmq_queue_length_monitor_millis, pos_integer),
+    {ok, TRef} = timer:send_interval(Interval, status_ping),
+    TRef.
+
+% this function just returns ok, as a return value is communicated back to
+% the gen_server via check_current_queue_length
+-spec check_current_queue_state(pid(), integer()) -> ok.
+check_current_queue_state(ParentPid, DroppedSinceLastCheck) ->
+    case get_max_length() of
+        undefined -> ok;
```

Comment:
the ParentPid doesn't need to be notified in this case, there are no results to add to the state